rasa/core/policies/rule_policy.py

```diff
@@ -286,11 +286,11 @@ def _check_slots_fingerprint(
     ) -> Set[Text]:
         expected_slots = set(fingerprint.get(SLOTS, {}))
         current_slots = set(state.get(SLOTS, {}).keys())
-        if expected_slots == current_slots:
-            # all expected slots are satisfied
+        if expected_slots.issubset(current_slots):
+            # all expected slots are found in the currently set slots
             return set()
 
-        return expected_slots
+        return expected_slots.difference(current_slots)
```

Comment:
can we simply return `expected_slots.difference(current_slots)`? in case of subset, it'll be exactly `set()`, no?

Reply:
You are right, should've done it right away.